api/base/filters.py

```diff
@@ -24,17 +24,30 @@
 def sort_multiple(fields):
     fields = list(fields)
     def sort_fn(a, b):
-        while fields:
-            field = fields.pop(0)
-            a_field = getattr(a, field)
-            b_field = getattr(b, field)
-            if a_field > b_field:
-                return 1
-            elif a_field < b_field:
-                return -1
+        for field in fields:
+            if field[0] == '-':
+                field = field[1:]
+                a_field = getattr(a, field)
+                b_field = getattr(b, field)
+                if a_field > b_field:
+                    return -1
+                elif a_field < b_field:
+                    return 1
+                elif a_field == b_field:
+                    return 1
```

Comment:
The (an?) error here is in sorting based on multiple fields; here's a field for a failing test (also illustrating the suggestion at https://github.com/CenterForOpenScience/osf.io/pull/5258#discussion_r57056259):

```
diff --git a/api_tests/base/test_filters.py b/api_tests/base/test_filters.py
index 7bcfefb..a9a4d4d 100644
--- a/api_tests/base/test_filters.py
+++ b/api_tests/base/test_filters.py
@@ -277,3 +277,15 @@ class TestODMOrderingFilter(ApiTestCase):
             sortt.append(str(i))
         assert_equal((sortt) , ['Zip', 'NewProj', 'Activity', 'Activity'])
 
+
+    class obj:
+        def __init__(self, **kw):
+            self.__dict__.update(**kw)
+
+    def test_filter_queryset_handles_multiple_fields(self):
+        objs = [self.obj(title='NewProj', n=10),
+                self.obj(title='Zip', n=20),
+                self.obj(title='Activity', n=30),
+                self.obj(title='Activity', n=40)]
+        actual = [x.n for x in sorted(objs, cmp=filters.sort_multiple(['title', '-number']))]
+        assert_equal(actual, [40, 30, 10, 20])
```